eZ/Publish/Core/MVC/Symfony/FieldType/ImageAsset/ParameterProvider.php

```diff
@@ -33,17 +46,52 @@ public function __construct(ContentService $contentService)
     public function getViewParameters(Field $field): array
     {
         try {
-            $contentInfo = $this->contentService->loadContentInfo(
+            $contentInfo = $this->loadContentInfo(
                 $field->value->destinationContentId
             );
 
-            return [
-                'available' => !$contentInfo->isTrashed(),
-            ];
+            if (!$this->userHasPermissions($contentInfo)) {
+                return [
+                    'available' => false,
+                ];
+            }
+
+            return ['available' => !$contentInfo->isTrashed()];
         } catch (NotFoundException | UnauthorizedException $exception) {
             return [
                 'available' => false,
             ];
         }
     }
+
+    /**
+     * @param int $id
+     * @return \eZ\Publish\API\Repository\Values\Content\ContentInfo
+     * @throws \Exception
+     */
+    private function loadContentInfo(int $id): ContentInfo
+    {
+        return  $this->repository->sudo(
+            function (Repository $repository) use ($id) {
+                return $repository->getContentService()->loadContentInfo($id);
+            }
+        );
+    }
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\ContentInfo $contentInfo
+     * @return bool
+     */
+    private function userHasPermissions(ContentInfo $contentInfo): bool
+    {
+        $readAttribute = new Attribute('content', 'read', [
+            'valueObject' => $contentInfo,
+        ]);
+
+        $viewEmbedAttribute = new Attribute('content', 'view_embed', [
+            'valueObject' => $contentInfo,
+        ]);
+
+        return $this->authorizationChecker->isGranted($readAttribute) || $this->authorizationChecker->isGranted($viewEmbedAttribute);
```

Comment:
@kmadejski will this hit the issue you solved in your PR, also if this is used outside of web context this might be an issue. Or is it desirable to use authorizationChecker here?

Reply:
In my opinion, `PermissionResolver` should be used here as well. `AuthorizationChecker` might throw an non existing token exception in the case when Symfony Security component has not been initialized yet. I can imagine a situation when a user wants to use an image on 404 pages for instance. Therefore, 
@pawbuj please check https://jira.ez.no/browse/EZP-29959 & https://github.com/ezsystems/ezpublish-kernel/pull/2510 for more info.